digdag-standards/src/main/java/io/digdag/standards/operator/EmbulkOperatorFactory.java

```diff
@@ -77,64 +87,118 @@ public EmbulkOperator(OperatorContext context)
+
+            final CommandStatus status;
+            if (!stateParams.has("commandStatus")) {
+                // Run the code since command state doesn't exist
+                status = runCommand(params, commandContext);
+            }
+            else {
+                // Check the status of the code running
+                final ObjectNode previousStatusJson = stateParams.get("commandStatus", ObjectNode.class);
+                status = exec.poll(commandContext, previousStatusJson);
+            }
+
+            if (status.isFinished()) {
+                final int statusCode = status.getStatusCode();
+                if (statusCode != 0) {
+                    // Remove the polling state after fetching the result so that the result fetch can be retried
+                    // without resubmitting the code.
+                    stateParams.remove("commandStatus");
+                    throw new RuntimeException("Command failed with code " + statusCode);
                 }
+                return;
             }
-            catch (IOException | TemplateException ex) {
-                throw Throwables.propagate(ex);
+            else {
+                stateParams.set("commandStatus", status);
+                throw TaskExecutionException.ofNextPolling(scriptPollInterval, ConfigElement.copyOf(stateParams));
             }
+        }
 
-            ProcessBuilder pb = new ProcessBuilder("embulk", "run", tempFile);
-            pb.directory(workspace.getPath().toFile());
-            pb.redirectErrorStream(true);
+        private CommandStatus runCommand(final Config params, final CommandContext commandContext)
+                throws IOException, InterruptedException, TemplateException
+        {
+            final Path tempDir = workspace.createTempDir(String.format("digdag-embulk-%d-", request.getTaskId()));
+            final Path workingDirectory = workspace.getPath(); // absolute
+            final Path embulkYmlPath = tempDir.resolve("load.yml");
 
-            int ecode;
-            try {
-                Process p = exec.start(workspace.getPath(), request, pb);
-                p.getOutputStream().close();
+            if (params.has("_command")) {
+                String configData = params.get("_command", String.class);
+                String data = workspace.templateFile(templateEngine, configData, UTF_8, params);
 
-                // copy stdout to System.out and logger
-                clog.copyStdout(p, System.out);
+                ObjectNode embulkConfig;
+                try {
+                    embulkConfig = new YamlLoader().loadString(data);
+                }
+                catch (RuntimeException | IOException ex) {
+                    Throwables.propagateIfInstanceOf(ex, ConfigException.class);
+                    throw new ConfigException("Failed to parse yaml file", ex);
+                }
 
-                ecode = p.waitFor();
+                Files.write(workingDirectory.relativize(embulkYmlPath),
```

Comment:
is `workingDirectory.relativize(embulkYmlPath)` valid? I think this should be just `embulkYmlPath`.

Reply:
Got it. thank you, fixed https://github.com/treasure-data/digdag/pull/835/commits/44306964308eba2983a9e1626368ac8720937391